cli/parse_pdfs.py

```diff
@@ -77,73 +107,83 @@ def parse_file(
     # TODO: do we want to handle exceptions raised by get_pdf here?
     with tempfile.TemporaryDirectory() as temp_output_dir:
         pdf_path = download_pdf(input_task, temp_output_dir)
-        page_layouts, pdf_images = lp.load_pdf(pdf_path, load_images=True)
-        document_md5sum = hashlib.md5(pdf_path.read_bytes()).hexdigest()
-
-    # FIXME: handle EmptyFileError here using _pdf_num_pages
-
-    model = _get_detectron_model(model, device)
-    if ocr_agent == "tesseract":
-        ocr_agent = lp.TesseractAgent()
-    elif ocr_agent == "gcv":
-        ocr_agent = lp.GCVAgent()
-
-    all_pages_metadata = []
-    all_text_blocks = []
-
-    for page_idx, image in tqdm(
-        enumerate(pdf_images), total=len(pdf_images), desc=pdf_path.name
-    ):
-        # Maybe we should always pass a layout object into the PageParser class.
-        layout_disambiguator = LayoutDisambiguator(
-            image, model, model_threshold_restrictive
-        )
-        initial_layout = layout_disambiguator.layout
-        if len(initial_layout) == 0:
-            logging.info(f"No layout found for page {page_idx}.")
-            continue
-        disambiguated_layout = layout_disambiguator.disambiguate_layout()
-        postprocessor = PostProcessor(disambiguated_layout)
-        ocr_blocks = postprocessor.postprocess()
-        ocr_processor = OCRProcessor(
-            image=np.array(image),
-            page_number=page_idx,
-            layout=ocr_blocks,
-            ocr_agent=ocr_agent,
-        )
-        page_text_blocks = ocr_processor.process_layout()
-        all_text_blocks += page_text_blocks
-
-        page_dimensions = (
-            page_layouts[page_idx].page_data["width"],
-            page_layouts[page_idx].page_data["height"],
-        )
-        page_metadata = PDFPageMetadata(
-            dimensions=page_dimensions,
-            page_number=page_idx,
-        )
-
-        all_pages_metadata.append(page_metadata)
 
-    document = ParserOutput(
-        document_id=input_task.document_id,
-        document_url=input_task.document_url,
-        document_name=input_task.document_name,
-        document_description=input_task.document_description,
-        document_content_type=input_task.document_content_type,
-        document_slug=input_task.document_slug,
-        pdf_data=PDFData(
-            page_metadata=all_pages_metadata,
-            md5sum=document_md5sum,
-            text_blocks=all_text_blocks,
-        ),
-    ).set_document_languages_from_text_blocks(min_language_proportion=0.4)
-
-    output_path = output_dir / f"{input_task.document_id}.json"
-
-    output_path.write_text(document.json(indent=4, ensure_ascii=False))
-
-    logging.info(f"Saved {output_path.name} to {output_dir}.")
+        if pdf_path is None:
+            logging.info(
+                f"PDF path is None for: {input_task.document_url} at {temp_output_dir}."
```

Comment:
it could be nice to explain why the pdf path is none rather than saying this? otherwise understanding the log message means opening up the code

Reply:
Updated error log statement.